File: japiwdbe/WdbeQSegList.java
/**
	* \file WdbeQSegList.h
	* Java API code for record of table TblWdbeQSegList
	* \copyright (C) 2018-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 26 Aug 2021
  */
// IP header --- ABOVE

package apiwdbe;

import org.w3c.dom.*;
import sbecore.*;

public class WdbeQSegList {
	
	public WdbeQSegList(
				int jnum
				, String sref
				, String stubPplRefWdbeMPipeline
				, String stubSupRefWdbeMSegment
				, String Infmt
				, String Outfmt
				, int Latency
			) {
		this.jnum = jnum;
		this.sref = sref;
		this.stubPplRefWdbeMPipeline = stubPplRefWdbeMPipeline;
		this.stubSupRefWdbeMSegment = stubSupRefWdbeMSegment;
		this.Infmt = Infmt;
		this.Outfmt = Outfmt;
		this.Latency = Latency;
	};

	public int jnum;
	public String sref;
	public String stubPplRefWdbeMPipeline;
	public String stubSupRefWdbeMSegment;
	public String Infmt;
	public String Outfmt;
	public int Latency;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WdbeQSegList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			stubPplRefWdbeMPipeline = Xmlio.extractStringUclc(doc, basexpath, "stubPplRefWdbeMPipeline", "ppl", null, 0);
			stubSupRefWdbeMSegment = Xmlio.extractStringUclc(doc, basexpath, "stubSupRefWdbeMSegment", "sup", null, 0);
			Infmt = Xmlio.extractStringUclc(doc, basexpath, "Infmt", "ifm", null, 0);
			Outfmt = Xmlio.extractStringUclc(doc, basexpath, "Outfmt", "ofm", null, 0);
			Latency = Xmlio.extractIntegerUclc(doc, basexpath, "Latency", "lcy", null, 0);
			
			return true;
		};

		return false;
	};
};

File: apiwdbe/WdbeQSegList.cpp
/**
	* \file WdbeQSegList.cpp
	* API code for table TblWdbeQSegList (implementation)
	* \copyright (C) 2016-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 26 Aug 2021
  */
// IP header --- ABOVE

#include "WdbeQSegList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WdbeQSegList
 ******************************************************************************/

WdbeQSegList::WdbeQSegList(
			const uint jnum
			, const string sref
			, const string stubPplRefWdbeMPipeline
			, const string stubSupRefWdbeMSegment
			, const string Infmt
			, const string Outfmt
			, const usmallint Latency
		) :
			jnum(jnum)
			, sref(sref)
			, stubPplRefWdbeMPipeline(stubPplRefWdbeMPipeline)
			, stubSupRefWdbeMSegment(stubSupRefWdbeMSegment)
			, Infmt(Infmt)
			, Outfmt(Outfmt)
			, Latency(Latency)
		{
};

bool WdbeQSegList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WdbeQSegList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "stubPplRefWdbeMPipeline", "ppl", stubPplRefWdbeMPipeline);
		extractStringUclc(docctx, basexpath, "stubSupRefWdbeMSegment", "sup", stubSupRefWdbeMSegment);
		extractStringUclc(docctx, basexpath, "Infmt", "ifm", Infmt);
		extractStringUclc(docctx, basexpath, "Outfmt", "ofm", Outfmt);
		extractUsmallintUclc(docctx, basexpath, "Latency", "lcy", Latency);
	};

	return basefound;
};

/******************************************************************************
 class ListWdbeQSegList
 ******************************************************************************/

ListWdbeQSegList::ListWdbeQSegList() {
};

ListWdbeQSegList::ListWdbeQSegList(
			const ListWdbeQSegList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQSegList(*(src.nodes[i]));
};

ListWdbeQSegList& ListWdbeQSegList::operator=(
			const ListWdbeQSegList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQSegList(*(src.nodes[i]));

	return *this;
};

ListWdbeQSegList::~ListWdbeQSegList() {
	clear();
};

void ListWdbeQSegList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWdbeQSegList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WdbeQSegList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWdbeQSegList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WdbeQSegList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WdbeQSegList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WdbeQSegList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};
